src/Shared/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
```

Comment:
I would leave this file under Authorization for now... it's only used there and is quite specific for the permissions...